buildbot/master/files/config/steps.yml

```diff
@@ -22,6 +22,12 @@ mac-rel-css:
   - bash ./etc/ci/lockfile_changed.sh
   - bash ./etc/ci/manifest_changed.sh
 
+mac-nightly:
+  - ./mach build --release
+  - ./mach package --release
+  - s3cmd -c ~/.s3cfg-servo put target/release/*.tar.gz s3://servo-developer-preview/nightly/`date +%Y`/
```

Comment:
Also the mac package script now generates a `.dmg` instead of a `.tar.gz` and it is put in the `target` directory (not `debug` or `release` directories)